acme/acme/jose/util.py

```diff
@@ -43,8 +43,17 @@ def __init__(self, wrapped):
     def __getattr__(self, name):
         return getattr(self._wrapped, name)
 
-    def _dump(self, filetype=OpenSSL.crypto.FILETYPE_ASN1):
-        # pylint: disable=missing-docstring,protected-access
+    def dump(self, filetype=OpenSSL.crypto.FILETYPE_ASN1):
```

Comment:
I don't think this API should be exposed on `ComparableX509`. For instance it leads to abuses such as the one from this very PR:
```python
text = jose.ComparableX509(cert_or_req).dump(OpenSSL.crypto.FILETYPE_TEXT)
```

Moreover, if we want to really expose this API (not saying here if we should), then `filetype` argument shall not have default value as it leads to less readable code: `.dump()` - is that PEM, DER or maybe text representation?